### PR TITLE
Fixed customizable reagent holders for held holders

### LINKED ISSUE #56544: Making custom foods while holding the ingredients in hand puts the food in your mob
## Round ID:
Tested locally

## Testmerges:
N/A

## Reproduction:
Hold a bowl
Hold a food item to make a salad out of too
click on bowl with food
it goes into your contents.

Works with burgers and such too.

image

### PR DESCRIPTION
**Pull request by swindly**
## About The Pull Request

Changes customizable_reagent_holder to put the new parent item in the hands of the attacker when the old parent is being held by the attacker.

## Why It's Good For The Game

Fixes https://github.com/tgstation/tgstation/issues/56544

## Changelog
:cl:
fix: Fixed tablecrafted foods being put inside their creators when they are held.
/:cl:

### CODE REVIEW

File: code/datums/components/customizable_reagent_holder.dm [hunk 1]
```diff
@@ -123,6 +123,11 @@
 		return
 	if (replacement)
 		var/atom/replacement_parent = new replacement(atom_parent.loc)
+		if(atom_parent.loc == attacker && isitem(replacement_parent) && isitem(atom_parent))
+			var/obj/item/I = atom_parent
```
Comment (Mothblocks) [inline code/datums/components/customizable_reagent_holder.dm:127]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/datums/components/customizable_reagent_holder.dm [hunk 2]
```diff
@@ -123,6 +123,11 @@
...
+			if(attacker.is_holding(I))
+				var/obj/item/R = replacement_parent
```
Comment (Mothblocks) [inline code/datums/components/customizable_reagent_holder.dm:129]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
Undraft your PR when it is ready (and passes linters).

File: code/datums/components/customizable_reagent_holder.dm [hunk 3]
```diff
@@ -123,6 +123,11 @@
 		return
 	if (replacement)
 		var/atom/replacement_parent = new replacement(atom_parent.loc)
+		if(atom_parent.loc == attacker && isitem(replacement_parent) && isitem(atom_parent))
+			var/obj/item/old_item = atom_parent
+			if(attacker.is_holding(old_item))
+				var/obj/item/replacement_item = replacement_parent
+				attacker.put_in_hand(replacement_item, attacker.get_held_index_of_item(old_item), TRUE)
```
Comment (Mothblocks) [inline code/datums/components/customizable_reagent_holder.dm:130]: Use a named argument for `TRUE`.

**Review by Mothblocks — APPROVED** [review]

**Review by Mothblocks — COMMENTED** [review]
wait mmm actually i dont know if i like that force

is there any reason you're not doing something like deleting/nullspacing the existing item? can i get an understanding at the thought process here?

**Comment by swindly** [comment]
The existing item needs to continue existing after the replacement item is created so that the component can be transferred. The existing item gets deleted a few lines later.

**Comment by Mothblocks** [comment]
It gives me pause when it comes to things like NODROP. I think you should instead call `dropItemToGround` or whatever the proc is directly, and handle the fail state of that.

**Comment by swindly** [comment]
Copying flags like NODROP to the replacement parent and adding conditions on which to fail to delete the old item will change the behavior of food tablecrafting.

**Pull request closed without merging**